UofT-Frosh-Orientation/orientation-website
# 641 implement backend logs

## **Contribution:**


### **Issue:**

Closes #641 

### **Accomplishments:**
- Created logger.js file in server/utils that define a pino logger to handle backend logging in express
- Writes logs to output file (currently just servers/utils/output.log)

### **Visuals:** 

![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/79293445/b179535a-7ee7-4534-9468-54a9bfd033d0)

## **Details:**

### **Expected Behavior:**

- Should log errors/information with varying levels (ranging from 'DEBUG' to 'FATAL', indicating the importance of the log)
- Logs important user events such as sign up, login, registration etc. (Working on it)
- Does not emit any Personal Identifiable Information or secret keys.

### **Testing Steps:**

Steps required to get expected behavior.

- Run the frontend + backend and signup, login, etc. (Use the website), and see the behavior of the logger and if it is logging correctly. It should provide logs for major user events, especially if the website crashes in any way.

### **Complications:**

- Was weird using pino vs pinoHTTP, until I figured out that you could use a pino logger as a parameter into the pinoHTTP logger.
- Some trouble with docker (mongo-express restarting constantly), still no proper solution other than restarting a few times until it works.

### **Resources:**

Any additional resources used outside this in issue.

- [resource 1](https://github.com/pinojs/pino/blob/master/docs/redaction.md) for Redacting information
